Monika After Story/game/cardgames.rpy

```diff
@@ -0,0 +1,5639 @@
+                # items format: {"type": type, "turn": turn_id, "monika_card": card or None, "player_card": card or None, "seen_count": int, "shown": boolean}
+                self.reactions = []
+
+            def __repr__(self):
+                return "<__AI Monika>"
+
+            def thonk_pause(self):
+                """
+                Pauses the game giving some time to Monika to thonk out her next turn
+                """
+                if len(self.hand) == 1:
+                    thonk_time = self.MIN_THONK_TIME
+
+                else:
+                    thonk_time = renpy.random.uniform(self.MIN_THONK_TIME, self.MAX_THONK_TIME)
+
+                renpy.pause(thonk_time, hard=True)
+
+            def __randomise_colour(self):
+                """
+                Chooses one of the colours at random
+                Excludes the potential colour that the player may have
+                If we know what the colours the player doesn't have,
+                we will return one of them at random
+
+                OUT:
+                    string with one of 4 colours
+                """
+                if self.player_cards_data["lacks_colours"]:
+                    return renpy.random.choice(self.player_cards_data["lacks_colours"])
+
+                colours = list(self.game.COLOURS)
+
+                if self.player_cards_data["has_colour"] is not None:
+                    colours.remove(self.player_cards_data["has_colour"])
+
+                return renpy.random.choice(colours)
+
+            def guess_player_cards(self):
+                """
+                Guesses cards' colours in the player's hand
+                NOTE: must run this before anything else
+                NOTE: this's method is quite a mess
+                """
+                # We can't assume anything if the player didn't play their 1st turn
+                if len(self.game.game_log) < 2:
+                    return
+
+                # Decrement the counter if the pleyer played a card
+                if (
+                    self.player_cards_data["reset_in"] > 0
+                    and self.game.game_log[-2]["played_card"] is not None
+                ):
+                    self.player_cards_data["reset_in"] -= 1
+
+                # First let's guess what colour the player may have
+                # when you set a new colour, you probably have cards with that colour
+                if (
+                    (
+                        self.game.game_log[-2]["played_card"] is not None
+                        and self.game.game_log[-2]["played_card"].type == "wild"
+                    )
+                    or (
+                        self.game.current_turn == 2
+                        and len(self.game.discardpile) > 1
+                        and self.game.discardpile[-2].type == "wild"
+                    )
+                ):
+                    colour = self.game.game_log[-2]["played_card"].colour
```

Comment:
I only noticed it now, but honestly speaking, for overall consistency with Python and other programming languages in general, we shouldn't use the British English spelling of `colour`, but we should be using `color` instead.